src/Microsoft.AspNet.Server.Testing/Deployers/ApplicationDeployer.cs

```diff
@@ -99,16 +99,31 @@ protected string DnxCommandName
 
         protected string PopulateChosenRuntimeInformation()
         {
-            // ex: runtimes/dnx-coreclr-win-x64.1.0.0-rc1-15844/bin
-            var currentRuntimeBinPath = Path.GetDirectoryName(Process.GetCurrentProcess().MainModule.FileName);
+            string currentRuntimeBinPath = string.Empty;
+            if (TestPlatformHelper.IsMac && DeploymentParameters.RuntimeFlavor == RuntimeFlavor.CoreClr)
+            {
+                var path = Environment.GetEnvironmentVariable("PATH");
+                currentRuntimeBinPath = path.Split(new char[] { ':' }, StringSplitOptions.RemoveEmptyEntries).
+                    Where(c => c.Contains("dnx-coreclr-darwin") || c.Contains("dnx-mono")).FirstOrDefault();
+
+                if (string.IsNullOrWhiteSpace(currentRuntimeBinPath))
+                {
+                    throw new Exception("Runtime not detected on the machine.");
+                }
+            }
+            else
+            {
+                // ex: runtimes/dnx-coreclr-win-x64.1.0.0-rc1-15844/bin
+                currentRuntimeBinPath = Path.GetDirectoryName(Process.GetCurrentProcess().MainModule.FileName);
```

Comment:
You need not have this condition any more as we can have one way of getting current runtime bin path and it could be used both on mono and regular deployers...
```
protected string getcurrentruntimepath()
{
            var path = Environment.GetEnvironmentVariable("PATH");
            var runtimeBin = path.Split(new char[] { ':' }, StringSplitOptions.RemoveEmptyEntries).
                Where(c => c.Contains("dnx-mono") || c.Contains("dnx-clr") || c.Contains("dnx-coreclr")).FirstOrDefault();

            if (string.IsNullOrWhiteSpace(runtimeBin))
            {
                throw new Exception("Runtime not detected on the machine.");
            }

     return path;
}
```